File: 49.group_anagrams/groupAnagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    // var sorted = {};
    var table = new Map();
    var result = [];
    for(var i = 0; i < strs.length; i++){
        var str = strs[i].split("").sort().join("");
        if(!table.get(str)){
            table.set(str,[strs[i]]);
        }else{
            var arr = table.get(str);
            arr.push(strs[i]);
            table.set(str,arr);
        }
    }
    for(let [key, value] of table){
        result.push(value);
    }
    return result;
    
};

File: 49.group_anagrams/groupAnagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        if not strs:
            return [[]]
        dic = dict()
        for i in strs:
            comb = "".join(sorted(i))
            if dic.get(comb):
                dic[comb].append(i)
            else:
                dic[comb] = [i]
        return dic.values()